### ISSUE TITLE
App crashes when evaluating `toIList` expression in IntelliJ or VSCode using debugger

### ISSUE THREAD

**Issue by saibotma** [marcglasberg/fast_immutable_collections#42]
Put a breakpoint on line 14 and execute the code below using IntelliJ in debug mode:

```dart
import 'package:fast_immutable_collections/fast_immutable_collections.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    final test = [1, 2].toIList();
    return Container();
  }
}
```

Execute `test` as expression in the “Variables” tab of the debugger window.
Notice that the app crashes without printing any exception.

![Screenshot 2022-07-07 at 22 42 23](https://user-images.githubusercontent.com/26022902/177867775-bea35205-a3a6-44d4-a3ae-62d6082ba0c4.png)

The same also happens when creating a watch on the `test` expression in VSCode.

Did crash on macOS, iPad, and Android. Did not test other platforms.
The same error also happens when calling `toISet`. Did not test with other methods.
Did not happen in Flutter < 3.x.x.

IntelliJ Version: 2022.1.3
Flutter Plugin Version: 69.0.4
Flutter Version: 3.0.4

**Comment by marcglasberg** [comment]
Thanks for catching this. There are 2 possibilities:

* The Dart debugger has a bug; or
* The `IList` class has a bug that is being triggered by the debugger. 

Or both.

I opened an issue here: https://github.com/dart-lang/sdk/issues/49430 and I'll need to wait until they answer so that I can fix this.

**Comment by sanderhyipyip** [comment]
I can't reproduce this anymore on Flutter `master`. Haven't tested any beta versions (yet). In our application switching from 3.0.5 to master currently does not seem to break anything. I guess it could be an okay-ish workaround just till the next Flutter stable.

**Comment by marcglasberg** [comment]
Ok. So closing this.

**Comment by saibotma** [comment]
Seems to be fixed with Dart 2.18 according to https://github.com/dart-lang/sdk/issues/49430.
